internal/logger/telemetry_store.go

```diff
@@ -64,6 +64,19 @@ type RuleEvalData struct {
 	// TODO: do we want to store params?
 }
 
+// ProjectMetadata can be used to store project metadata in the context.
+type ProjectMetadata struct {
```

Comment:
Can we call this "ProjectTombstone" or "ProjectDeletion", so it's clear that this is specifically for the case when a project is deleted?

Reply:
Thanks for catching that and my bad for the obvious miss!

I've added the log audit at the core function which seems to be called from all other paths.